### DISCUSSION TITLE
spacy multilabel text classification changing loss value, doesn't change result

### DISCUSSION THREAD

**Discussion by darwinharianto** [explosion/spaCy#7110] [Help: Coding & Implementations]
I was trying to make a prediction on a text with multi class annotations.

This is my config for training
```
[paths]
train = "data_ja/train.spacy"
dev = "data_ja/valid.spacy"
vectors = null
init_tok2vec = null

[system]
gpu_allocator = null
seed = 0

[nlp]
lang = "en"
pipeline = ["tok2vec","textcat_multilabel"]
batch_size = 200
disabled = []
before_creation = null
after_creation = null
after_pipeline_creation = null
tokenizer = {"@tokenizers":"spacy.Tokenizer.v1"}

[components]

[components.textcat_multilabel]
factory = "textcat_multilabel"
threshold = 0.5

[components.textcat_multilabel.model]
@architectures = "spacy.TextCatEnsemble.v2"
nO = null

[components.textcat_multilabel.model.linear_model]
@architectures = "spacy.TextCatBOW.v1"
exclusive_classes = false
ngram_size = 1
no_output_layer = false
nO = null

[components.textcat_multilabel.model.tok2vec]
@architectures = "spacy.Tok2VecListener.v1"
width = ${components.tok2vec.model.encode.width}
upstream = "*"

[components.tok2vec]
factory = "tok2vec"

[components.tok2vec.model]
@architectures = "spacy.Tok2Vec.v2"

[components.tok2vec.model.embed]
@architectures = "spacy.MultiHashEmbed.v1"
width = ${components.tok2vec.model.encode.width}
attrs = ["ORTH","SHAPE"]
rows = [5000,2500]
include_static_vectors = false

[components.tok2vec.model.encode]
@architectures = "spacy.MaxoutWindowEncoder.v2"
width = 96
depth = 4
window_size = 1
maxout_pieces = 3

[corpora]

[corpora.dev]
@readers = "spacy.Corpus.v1"
path = ${paths.dev}
max_length = 0
gold_preproc = false
limit = 0
augmenter = null

[corpora.train]
@readers = "spacy.Corpus.v1"
path = ${paths.train}
max_length = 2000
gold_preproc = false
limit = 0
augmenter = null

[training]
dev_corpus = "corpora.dev"
train_corpus = "corpora.train"
seed = ${system.seed}
gpu_allocator = ${system.gpu_allocator}
dropout = 0.1
accumulate_gradient = 1
patience = 5000
max_epochs = 0
max_steps = 100000
eval_frequency = 200
frozen_components = []
before_to_disk = null

[training.batcher]
@batchers = "spacy.batch_by_words.v1"
discard_oversize = false
tolerance = 0.2
get_length = null

[training.batcher.size]
@schedules = "compounding.v1"
start = 100
stop = 1000
compound = 1.001
t = 0.0

[training.logger]
@loggers = "spacy.ConsoleLogger.v1"
progress_bar = false

[training.optimizer]
@optimizers = "Adam.v1"
beta1 = 0.9
beta2 = 0.999
L2_is_weight_decay = true
L2 = 0.01
grad_clip = 1.0
use_averages = false
eps = 0.00000001
learn_rate = 0.001

[training.score_weights]
cats_score_desc = null
cats_micro_p = null
cats_micro_r = null
cats_micro_f = null
cats_macro_p = 1.0
cats_macro_r = 1.0
cats_macro_f = 1.0
cats_macro_auc = 1.0
cats_f_per_type = null
cats_macro_auc_per_type = null
cats_score = 1.0

[pretraining]

[initialize]
vectors = null
init_tok2vec = ${paths.init_tok2vec}
vocab_data = null
lookups = null
before_init = null
after_init = null

[initialize.components]

[initialize.tokenizer]
```

When i train, there is no improvment on loss, it always stop after 1600 iteration (patience).

when i see the printed out message, i noticed that loss is always so low
```
E    #       LOSS TOK2VEC  LOSS TEXTC...  CATS_SCORE  CATS_MACRO_P  CATS_MACRO_R  CATS_MACRO_F  CATS_MACRO...  SCORE 
---  ------  ------------  -------------  ----------  ------------  ------------  ------------  -------------  ------
  0       0          0.01           0.00       46.21          6.71         40.99         10.43          46.21    0.30
 32     200          0.09           0.10       45.04         11.71          3.48          4.86          45.04    0.22
 72     400          0.01           0.01       44.68         10.10          3.26          4.62          44.68    0.21
122     600          0.00           0.01       45.13          9.78          2.57          3.74          45.13    0.21
182     800          0.00           0.01       44.71         14.45          3.72          5.49          44.71    0.23
248    1000          0.00           0.00       43.99         13.78          3.57          5.36          43.99    0.22
339    1200          0.00           0.00       43.74         13.37          3.57          5.24          43.74    0.22
439    1400          0.00           0.00       43.49         12.29          3.22          4.76          43.49    0.21
539    1600          0.00           0.00       44.36         12.86          3.43          5.14          44.36    0.22

```
So i tried to make the loss bigger by using tensorflow KL loss. 

New loss result:
```
E    #       LOSS TOK2VEC  LOSS TEXTC...  CATS_SCORE  CATS_MACRO_P  CATS_MACRO_R  CATS_MACRO_F  CATS_MACRO...  SCORE 
---  ------  ------------  -------------  ----------  ------------  ------------  ------------  -------------  ------
  0       0          0.01         214.97       46.21          6.71         40.99         10.43          46.21    0.30
 32     200          0.09        2770.75       45.04         11.71          3.48          4.86          45.04    0.22
 72     400          0.01         477.44       44.68         10.10          3.26          4.62          44.68    0.21
122     600          0.00         243.78       45.13          9.78          2.57          3.74          45.13    0.21
182     800          0.00         167.14       44.71         14.45          3.72          5.49          44.71    0.23
248    1000          0.00         125.89       43.99         13.78          3.57          5.36          43.99    0.22
339    1200          0.00          97.41       43.74         13.37          3.57          5.24          43.74    0.22
439    1400          0.00          75.10       43.49         12.29          3.22          4.76          43.49    0.21
539    1600          0.00          66.12       44.36         12.86          3.43          5.14          44.36    0.22
709    1800          0.00          52.18       43.20         13.12          3.57          5.23          43.20    0.22
909    2000          0.00          45.80       43.51         12.99          3.43          5.13          43.51    0.22
1109    2200          0.00          40.15       43.68         12.79          3.57          5.21          43.68    0.22

```
Even if i changed the loss, seems like result is identical. Is the loss used for classification is different than what is displayed?

For single class classification, everything works fine, Score could increase from 0.3 up to 0.5
